### PR TITLE
sctest: Skip "crdb_internal" entry in comparator testing from logictest corpus

### PR DESCRIPTION
**Pull request by Xiang-Gu**
Statements collected from `crdb_internal` is of little interest to us
for the purpose of schema changer comparator testings, as it contains
only `crdb_internal` functions. Furthermore, there are statements that
will crash our framework, such as `SELECT crdb_internal.force_panic('foo')`.

This commit therefore adds a check to skip it. This check will also home
any future skipped entries for whatever reason.

Informs #111978
Release note: None

### CODE REVIEW

File: pkg/sql/schemachanger/schemachanger_test.go [hunk 1]
```diff
@@ -913,13 +913,33 @@ func TestComparatorFromLogicTests(t *testing.T) {
 	for _, entry := range corpus.Entries {
 		subtestName := entry.Name
 		subtestStatements := entry.Statements
-		t.Run(entry.Name, func(t *testing.T) {
+
+		if shouldSkipLogicTestCorpusEntry(subtestName) {
+			continue
```
Comment (chrisseto) [inline pkg/sql/schemachanger/schemachanger_test.go:918]: Would it be helpful to still call `t.Run` and then `t.Skip` for these cases to show that they're captured correctly?

If we don't want to generate skipped tests, maybe just a `t.Log` that we're skipping this?

File: pkg/sql/schemachanger/schemachanger_test.go [hunk 2]
```diff
@@ -913,13 +913,33 @@ func TestComparatorFromLogicTests(t *testing.T) {
...
+		if !t.Run(entry.Name, func(t *testing.T) {
 			t.Logf("running schema changer comparator testing on statements collected from logic test %q\n", subtestName)
 			ss := &staticSQLStmtLineProvider{
 				stmts: subtestStatements,
 			}
 			sctest.CompareLegacyAndDeclarative(t, ss)
 			t.Logf("schema changer comparator testing succeeded on logic test %q\n", subtestName)
-		})
+		}) {
+			t.Logf("test failed on %q\n", subtestName)
```
Comment (chrisseto) [inline pkg/sql/schemachanger/schemachanger_test.go:929]: I'm a bit confused about this log statement. Won't we already have the information about what test failed from `go test`s output?

**Review by chrisseto — APPROVED** [review]
LGTM, a couple questions but nothing blocking!

File: pkg/sql/schemachanger/schemachanger_test.go (hunk 2 above)
Comment (Xiang-Gu) [reply pkg/sql/schemachanger/schemachanger_test.go:929]: I am having a bit trouble understanding the TC behavior as well. See this ongoing slack [thread](https://cockroachlabs.slack.com/archives/CJ0H8Q97C/p1696862216857819). I think we don't need it -- it's just this `crdb_internal` test case is giving us some extra hard time. 

Let me remove them and see.

File: pkg/sql/schemachanger/schemachanger_test.go (hunk 1 above)
Comment (Xiang-Gu) [reply pkg/sql/schemachanger/schemachanger_test.go:918]: I actually liked the idea to still call `t.Run` and skip inside it! I will change it

**Comment by Xiang-Gu** [comment]
TFTR!

bors r+

**Comment by yuzefovich** [comment]
This has red CI.

bors r-

**Comment by Xiang-Gu** [comment]
license/cla check is stuck. Force-push and eat another CI to fix it. 

bors r-

**Comment by Xiang-Gu** [comment]
blathers backport 23.2

**Pull request merged**